### PR TITLE
Move away from rata die reference and use iso_days instead

### PR DESCRIPTION
**Pull request by josevalim**
Move away from rata die reference and use iso_days instead

We chose to use the year zero in the astronomical year
numbering as a reference point, as it is also the reference
used by Calendar.ISO, the default calendar used by Elixir,

It is also the same epoch used by Erlang's :calendar module.
Note we chose to not call it gregorian epoch due to the confusion
between January 1 AC 1 and 0000-01-01. The latter actually maps
to January 1 BC 1.

We also moved away from Calendar.Julian, which could have been
based on Calendrical Calculations, and chose Calendar.Holocene,
which adds exactly 10,000 years to Calendar.ISO and as such has
a very straigh-forward implementation.

More information:

  + https://en.wikipedia.org/wiki/Year_zero
  + https://en.wikipedia.org/wiki/Holocene_calendar

### CODE REVIEW

File: lib/elixir/test/elixir/fixtures/calendar/holocene.exs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+defmodule Calendar.Holocene do
```
Comment (wojtekmach) [inline lib/elixir/test/elixir/fixtures/calendar/holocene.exs]: would it make sense to add `@behaviour Calendar` and `@impl true` for this one? This way it's a more complete implementation for people to look at and base off of. Also, would probably be good to use `@impl true` for `Calendar.ISO`.

If any of this sounds good I'm happy to prepare PR, either against this branch or after this lands.

Comment (josevalim) [inline lib/elixir/test/elixir/fixtures/calendar/holocene.exs]: @wojtekmach please do send a PR! That's actually what I tried to do but we don't implement all callbacks yet and getting this out was more important, so I postponed it. :)

File: lib/elixir/lib/calendar/datetime.ex [hunk 2]
```diff
@@ -580,15 +580,17 @@ defmodule DateTime do
 
   ## Examples
 
-  Imagine someone implements `Calendar.Julian`:
+  Imagine someone implements `Calendar.Holecene`, which is a calendar
+  based on the Gregorian calendar that adds exactly 10,000 years to the
+  current year:
```
Comment (Qqwy) [inline lib/elixir/lib/calendar/datetime.ex]: > Imagine someone implements `Calendar.Holocene` which is a calendar based on the Gregorian calendar that adds exactly 10,000 years to the current Gregorian year.

**Comment by Qqwy** [comment]
To be honest, I find the name `zero_year` meaning a number of days since year zero misleading. Is there another name we could use instead? Maybe `astronomical_days`?

The rest looks great! Using Holocene instead of Julian is a great idea to make the test example more clear.

**Comment by josevalim** [comment]
Thanks @Qqwy for the review and the mailing list reply.

I like astronomical_days but I am not that familiar with the astronomical field to say if that can mean something completely different. Along the same line though, what if we call it `iso_days`? It mirrors with the main calendar and representations and is exactly the same format we have today.

**Comment by josevalim** [comment]
I have renamed zero_year to iso_days and addressed comments. :heart:

**Pull request merged**